### ISSUE TITLE
Implement WallLoss Runnable Process for Aerosol Integration

### ISSUE THREAD

**Issue by Gorkowski** [uncscode/particula#819]
**Parent Issue:** #72

**Dependencies:**
- #816 - WallLossStrategy ABC and SphericalWallLossStrategy
- #817 - RectangularWallLossStrategy
- #818 - Wall Loss Builders and Factory

---

## Problem/Motivation

With the wall loss strategies (#816, #817) and builders (#818) implemented, we need to create the `WallLoss` runnable process class. This class wraps wall loss strategies and provides the `RunnableABC` interface for integration with the `Aerosol` class, enabling wall loss to be combined with other processes (condensation, coagulation) in simulation pipelines.

**Parent Issue:** #72 - Implement Wall Loss Strategies and Builders for All Particle Representations

**Dependencies:**
- #816 - WallLossStrategy ABC and SphericalWallLossStrategy
- #817 - RectangularWallLossStrategy
- #818 - Wall Loss Builders and Factory

**Value:**
- Enables wall loss as a runnable process in simulation pipelines
- Provides consistent interface matching `MassCondensation` and `Coagulation` classes
- Supports process chaining with the `|` operator
- Integrates with `Aerosol` object for complete chamber simulations

### Core Implementation
- [ ] Add `WallLoss` class to `particula/dynamics/particle_process.py`
- [ ] Inherit from `RunnableABC`
- [ ] Implement `execute(aerosol, time_step, sub_steps)` method
- [ ] Implement `rate(aerosol)` method
- [ ] Support all wall loss strategies (spherical, rectangular)

### Integration
- [ ] Update `particula/dynamics/__init__.py` to export `WallLoss`
- [ ] Ensure compatibility with process chaining (`|` operator)
- [ ] Update `particula/dynamics/wall_loss/__init__.py` if needed

### Testing
- [ ] Add tests to `particula/dynamics/tests/` or existing test file
- [ ] Test `execute()` reduces particle concentrations
- [ ] Test `rate()` returns correct loss rates
- [ ] Test process chaining with condensation and coagulation
- [ ] Test with both spherical and rectangular strategies
- [ ] Test with all distribution types

### Documentation
- [ ] Google-style docstrings with examples
- [ ] Examples showing integration with Aerosol

## Technical Context

**File to Modify:**
- `particula/dynamics/particle_process.py` (+80 LOC)

**Files to Modify for Exports:**
- `particula/dynamics/__init__.py` - Add WallLoss export

**Test File:**
- Create or extend tests in `particula/dynamics/tests/`

**Key Dependencies:**
- `particula/runnable.py` - RunnableABC base class
- `particula/aerosol.py` - Aerosol class
- Wall loss strategies from #816, #817

**Related Code (Pattern References):**
- `particula/dynamics/particle_process.py:22-123` - `MassCondensation` class (exact pattern to follow)
- `particula/dynamics/particle_process.py:126-223` - `Coagulation` class (exact pattern to follow)

### WallLoss Class (in particle_process.py)

```python
from particula.dynamics.wall_loss.wall_loss_strategies import WallLossStrategy


class WallLoss(RunnableABC):
    """Handles the wall loss process for aerosols in chambers.
    
    This class applies a specified wall loss strategy to particles
    in an Aerosol, reducing particle concentrations due to deposition
    onto chamber walls. It supports both spherical and rectangular
    chamber geometries.
    
    Attributes:
        wall_loss_strategy: The wall loss strategy used for calculating
            deposition rates.
    
    Methods:
        execute: Apply wall loss over a specified time step.
        rate: Calculate the wall loss rate for particles.
    
    Examples:
        ```python
        import particula as par
        
        # Create wall loss strategy
        strategy = par.dynamics.SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        
        # Create wall loss process
        wall_loss = par.dynamics.WallLoss(wall_loss_strategy=strategy)
        
        # Apply to aerosol
        updated_aerosol = wall_loss.execute(aerosol, time_step=1.0)
        
        # Get loss rates
        rates = wall_loss.rate(aerosol)
        ```
    
    References:
        - Crump, J. G., & Seinfeld, J. H. (1981). Turbulent deposition 
          and gravitational sedimentation of an aerosol in a vessel of 
          arbitrary shape. Journal of Aerosol Science, 12(5), 405-415.
    """
    
    def __init__(self, wall_loss_strategy: WallLossStrategy):
        """Initialize the WallLoss process.
        
        Args:
            wall_loss_strategy: The wall loss strategy to use,
                responsible for calculating wall deposition rates.
        
        Examples:
            ```python
            strategy = SphericalWallLossStrategy(
                wall_eddy_diffusivity=0.001,
                chamber_radius=0.5,
            )
            wall_loss = WallLoss(wall_loss_strategy=strategy)
            ```
        """
        self.wall_loss_strategy = wall_loss_strategy
    
    def execute(
        self, aerosol: Aerosol, time_step: float, sub_steps: int = 1
    ) -> Aerosol:
        """Apply the wall loss process over a given time step.
        
        Updates particle concentrations in the aerosol based on
        wall deposition rates calculated by the strategy.
        
        Args:
            aerosol: The Aerosol instance to modify.
            time_step: The total time interval for wall loss [s].
            sub_steps: Number of subdivisions for iterative calculation.
                Using more sub_steps improves accuracy for large time steps.
        
        Returns:
            The updated aerosol object after wall loss.
        
        Examples:
            ```python
            # Simple execution
            updated_aerosol = wall_loss.execute(aerosol, time_step=1.0)
            
            # With sub-steps for better accuracy
            updated_aerosol = wall_loss.execute(
                aerosol, time_step=10.0, sub_steps=10
            )
            ```
        """
        for _ in range(sub_steps):
            aerosol.particles = self.wall_loss_strategy.step(
                particle=aerosol.particles,
                temperature=aerosol.atmosphere.temperature,
                pressure=aerosol.atmosphere.total_pressure,
                time_step=time_step / sub_steps,
            )
        return aerosol
    
    def rate(self, aerosol: Aerosol) -> Any:
        """Compute wall loss rates for the aerosol particles.
        
        Returns the instantaneous rate of particle concentration change
        due to wall deposition, in units of #/m³·s.
        
        Args:
            aerosol: The Aerosol instance containing particles.
        
        Returns:
            Array of wall loss rates for each particle size,
            in units of #/m³·s (negative values indicate loss).
        
        Examples:
            ```python
            rates = wall_loss.rate(aerosol)
            print(f"Loss rates: {rates} #/m³·s")
            # Output: Loss rates: [-1.2e8, -3.4e7, ...] #/m³·s
            ```
        """
        return self.wall_loss_strategy.rate(
            particle=aerosol.particles,
            temperature=aerosol.atmosphere.temperature,
            pressure=aerosol.atmosphere.total_pressure,
        )
```

### Update Imports (in particle_process.py)

```python
# Add to imports at top of file
from particula.dynamics.wall_loss.wall_loss_strategies import WallLossStrategy
```

### Update dynamics __init__.py

```python
# Add to particula/dynamics/__init__.py
from particula.dynamics.particle_process import (
    MassCondensation,
    Coagulation,
    WallLoss,  # Add this
)

# Update __all__ to include WallLoss
```

## Suggested Tests

```python
"""Tests for WallLoss runnable process."""

import numpy as np
import pytest

from particula.dynamics.particle_process import WallLoss
from particula.dynamics.wall_loss.wall_loss_strategies import (
    SphericalWallLossStrategy,
    RectangularWallLossStrategy,
)
# Import Aerosol and builders for fixtures


class TestWallLoss:
    """Tests for WallLoss runnable process."""
    
    def test_init_with_spherical_strategy(self):
        """Test initialization with spherical strategy."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        assert wall_loss.wall_loss_strategy is strategy
    
    def test_init_with_rectangular_strategy(self):
        """Test initialization with rectangular strategy."""
        strategy = RectangularWallLossStrategy(
            wall_eddy_diffusivity=0.0001,
            chamber_dimensions=(1.0, 0.5, 0.5),
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        assert wall_loss.wall_loss_strategy is strategy
    
    def test_execute_reduces_concentration(self, aerosol_fixture):
        """Test that execute reduces particle concentrations."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        initial_conc = aerosol_fixture.particles.get_concentration().copy()
        updated_aerosol = wall_loss.execute(
            aerosol_fixture, time_step=1.0
        )
        
        assert np.all(
            updated_aerosol.particles.get_concentration() <= initial_conc
        )
    
    def test_execute_with_sub_steps(self, aerosol_fixture):
        """Test execute with sub-steps."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        # Should not raise and should reduce concentration
        updated = wall_loss.execute(
            aerosol_fixture, time_step=10.0, sub_steps=10
        )
        assert updated is not None
    
    def test_rate_returns_negative_values(self, aerosol_fixture):
        """Test that rate returns negative values (loss)."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        rates = wall_loss.rate(aerosol_fixture)
        assert np.all(rates <= 0)
    
    def test_rate_shape_matches_particles(self, aerosol_fixture):
        """Test that rate shape matches particle concentration shape."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        rates = wall_loss.rate(aerosol_fixture)
        expected_shape = aerosol_fixture.particles.get_concentration().shape
        assert rates.shape == expected_shape
    
    def test_process_chaining_with_coagulation(
        self, aerosol_fixture, coagulation_fixture
    ):
        """Test process chaining with coagulation using | operator."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        # Chain processes
        combined = coagulation_fixture | wall_loss
        
        # Execute combined process
        updated = combined.execute(aerosol_fixture, time_step=1.0)
        assert updated is not None
    
    def test_process_chaining_with_condensation(
        self, aerosol_fixture, condensation_fixture
    ):
        """Test process chaining with condensation using | operator."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        # Chain processes
        combined = condensation_fixture | wall_loss
        
        # Execute combined process
        updated = combined.execute(aerosol_fixture, time_step=1.0)
        assert updated is not None
    
    def test_with_rectangular_strategy(self, aerosol_fixture):
        """Test WallLoss with rectangular strategy."""
        strategy = RectangularWallLossStrategy(
            wall_eddy_diffusivity=0.0001,
            chamber_dimensions=(1.0, 0.5, 0.5),
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        updated = wall_loss.execute(aerosol_fixture, time_step=1.0)
        rates = wall_loss.rate(aerosol_fixture)
        
        assert updated is not None
        assert np.all(rates <= 0)
    
    def test_zero_time_step_no_change(self, aerosol_fixture):
        """Test that zero time step produces no change."""
        strategy = SphericalWallLossStrategy(
            wall_eddy_diffusivity=0.001,
            chamber_radius=0.5,
        )
        wall_loss = WallLoss(wall_loss_strategy=strategy)
        
        initial_conc = aerosol_fixture.particles.get_concentration().copy()
        updated = wall_loss.execute(aerosol_fixture, time_step=0.0)
        
        np.testing.assert_array_equal(
            updated.particles.get_concentration(),
            initial_conc
        )
```

## Success Criteria

- [ ] `WallLoss` class inherits from `RunnableABC`
- [ ] `execute()` correctly reduces particle concentrations
- [ ] `rate()` returns array of loss rates (negative values)
- [ ] Works with both spherical and rectangular strategies
- [ ] Process chaining with `|` operator works correctly
- [ ] Sub-steps parameter improves accuracy for large time steps
- [ ] All tests pass with >95% coverage
- [ ] Docstrings follow Google style with examples
- [ ] Exported from `particula.dynamics` module

## Edge Cases and Considerations

- **Zero time step**: Should produce no change in concentrations
- **Very large time step**: May produce negative concentrations (clamp to zero?)
- **Sub-steps**: More sub-steps = better accuracy but slower
- **Process chaining order**: Wall loss typically applied after other processes
- **Temperature/pressure**: Accessed from aerosol.atmosphere

## Example Usage

```python
import numpy as np
import particula as par

# Build aerosol with particles and atmosphere
aerosol = (
    par.AerosolBuilder()
    .add_particle(
        par.particles.PresetParticleRadiusBuilder()
        .set_mode(np.array([100e-9]))
        .set_geometric_standard_deviation(np.array([1.5]))
        .set_number_concentration(np.array([1e10]))
        .set_density(1000.0)
        .build()
    )
    .set_atmosphere(
        par.gas.AtmosphereBuilder()
        .set_temperature(298.0)
        .set_pressure(101325.0)
        .build()
    )
    .build()
)

# Create wall loss process
wall_loss_strategy = par.dynamics.SphericalWallLossStrategy(
    wall_eddy_diffusivity=0.001,
    chamber_radius=0.5,
)
wall_loss = par.dynamics.WallLoss(wall_loss_strategy=wall_loss_strategy)

# Get instantaneous loss rate
print(f"Initial concentration: {aerosol.particles.get_concentration()}")
rates = wall_loss.rate(aerosol)
print(f"Wall loss rate: {rates} #/m³·s")

# Simulate 100 seconds with 1-second time steps
for t in range(100):
    aerosol = wall_loss.execute(aerosol, time_step=1.0)

print(f"Final concentration: {aerosol.particles.get_concentration()}")

# Combine with coagulation
coag_strategy = par.dynamics.BrownianCoagulationStrategy(
    distribution_type="discrete"
)
coagulation = par.dynamics.Coagulation(coagulation_strategy=coag_strategy)

# Chain processes
combined_process = coagulation | wall_loss

# Run combined simulation
for t in range(100):
    aerosol = combined_process.execute(aerosol, time_step=1.0)
```

## References

**Parent Issue:**
- #72 - Implement Wall Loss Strategies and Builders

**Dependencies:**
- #816 - WallLossStrategy ABC and SphericalWallLossStrategy
- #817 - RectangularWallLossStrategy
- #818 - Wall Loss Builders and Factory

**Related Code:**
- `particula/dynamics/particle_process.py:22-123` - MassCondensation (pattern to follow)
- `particula/dynamics/particle_process.py:126-223` - Coagulation (pattern to follow)
- `particula/runnable.py` - RunnableABC base class
- `particula/aerosol.py` - Aerosol class

**Coding Standards:**
- `docs/Agent/code_style.md` - Python standards
- `docs/Agent/testing_guide.md` - Testing patterns
